SolarEnergy/Aero/aero.py

```diff
@@ -0,0 +1,225 @@
+        '''Records the status of the plane at the end of each iteration. (nb. will not include last WP & start.)'''
+        storeit = [self.time, self.speed, self.position.lon, self.position.lat, self.position.alt, self.irrad, 
+                   self.eloss, self.egain, self.currentbatt, round(100.0*self.currentbatt/self.maxbattpack, 3)]
+        self.record.append(storeit)
+
+    def RecordPerformance(self):
+        '''Records the performance curves of the aircraft, including stall/max speeds and climb info'''
+        heightl = numpy.linspace(0.0, 11000.0, 110)
+        speedl = numpy.linspace(15.0, 115.0, 100)
+        for height in heightl:
+            sthtl = []
+            sthtl.append(height)
+            sthtl.append(self.AirDensity(height)*self.convert.get('slugspft32kgpm3'))   # Convert dens. to SI for output.
+            sthtl.append(self.T(h=height)-273.15)     # 273.15 converts to Centigrade
+            sthtl.append(self.GetStall(height))
+            sthtl.append(self.GetMaxVelocityBounds(scl=self.maxpower, h=height))
+            tspdl = []
+            for spd in speedl:
+                tmpl = []
+                tmpl.append(spd)
+                tmpl.append(self.maxpower)
+                tmpl.append(self.PowerReqHP(Vf=spd*self.convert.get('mph2fps'), h=height)) # KJS wip
+                tspdl.append(tmpl)
+            sthtl.append(tspdl)
+            self.recordPD.append(sthtl)
+
+    def GetIrrad(self, pIrDataObj, datetime):
+        '''Get Irrd from nearest co-ord/time. pIrDataObj is a combined object (see irradiance.py)'''
+        ir = pIrDataObj.GetIrradiance(self.position, datetime)
+        self.irrad = ir
+        print("Returning GetIrrad", ir)
+        return ir
+
+    def _SetAspectRatio(self):
+        '''Returns the aspect ratio = Wingspan**2/Wing area (both are in ft)'''
+        self.AR = self.b**2/self.S
+        print("Aspect Ratio is ", self.AR)
+
+    def _GVB_ftn(self, Vf, scl, h=0.0):
+        '''Internal function used to evaluate roots of the power curve (Newton-Raphson method)'''
+        '''Units (mix of Imp. & SI): velocity in fps, scl (power) in hp, h(eight) in m'''
+        return self.PowerReqHP(Vf, h)-scl
+
+    def GetMaxVelocityBounds(self, scl=70.0, h=0.0): # Fix height dependence donem scl is max power ... neaten this
+        '''Returns the maximum speed in mph (from the power-curve). scl is the max engine power (hp), h in meters'''
+        # P(vf) - scl = PowerReqHP(self, Vf) -scl  # hcl & h are fixed params in the root finding
+        v2 = optimize.newton(self._GVB_ftn, 100.0, args=(scl, h,))  # 100.0 is a start estimate for the root.
+        print("v2 = ", v2*self.convert.get('fps2mph')) # 0.681 is fts-1 -> mph
+        return v2*self.convert.get('fps2mph')
+
+    def GetStall(self, h=0.0):
+        '''Returns the stall speed in mph. height in metres'''
+        vs = math.sqrt(2.0*self.weight/(self.AirDensity(h)*self.S*1.5)) # CL_max to 1.5 ? (1-2,  with flaps - no idea ?)
```

Comment:
No idea....??

Reply:
Long version :  CL_max is the maximum lift coefficient - which defines when a wing stalls. Short of a wind-tunnel or running fluid-dynamics sims I've not idea what exactly this is.
Short version : Rough guess probably isn't too far off the mark for this plane & comment edited :-)  !